File: Data Structures & Algorithms/binary-search/submission-8.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        int index;
        while(l <= r ){
            int index = (l+r) / 2;
            if(nums[index] == target) return index;
            if(nums[index] > target)
                r = index - 1;
            else
                l = index + 1;
        }    

        return -1;
    }
};

File: Data Structures & Algorithms/single-element-in-a-sorted-array/submission-8.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        l, r = 0, len(nums) - 1
        index: int
        while l <= r:
            index = l + (r-l) // 2
            if l == r:
                break

            offset = index
            offset -= 1 if index % 2 == 1 else 0
            if nums[offset] == nums[offset + 1]:
                l = offset + 2
            else:
                r = index - 1
        return nums[index]